agent/consul/leader_connect_ca.go

```diff
@@ -100,13 +100,6 @@ func NewCAManager(delegate caServerDelegate, leaderRoutineManager *routine.Manag
 	}
 }
 
-func (c *CAManager) reset() {
-	c.state = caStateUninitialized
-	c.primaryRoots = structs.IndexedCARoots{}
-	c.actingSecondaryCA = false
```

Comment:
TODO: check to see if this should be done in `Stop`, should primaryRoots and actingSecondarCA be called to zero things out on leader loss.

Reply:
I checked and I'm pretty sure it's needed to clean up the struc and the state. I added them as part of the stop

Reply:
How was `reset()` being used originally? I can't find where it was being called from

Reply:
It wasn't, which is why 7dae65c was able to remove the method without any other changes. While reviewing this I thought it might be worth checking to see if the `Stop` method was missing any of this logic, and I guess it was.

I believe an earlier change must have removed the call to `reset`, but I haven't looked into it.

Reply:
`reset` seems like it was added in 0bfda4481f004fd40759b9afe99518dc5b37bc3a (#9009) but never used.

Reply:
Got it. I just saw that when reset was added it wasn't being called then either, looks like a stray change.

The important parts of `reset` were to set the state to uninitialized and set the provider to nil, and those were being handled separately in `revokeLeadership`.

It's definitely clearer to move those down into caManager.Stop() rather than spread around.